specification/dimensions/commitmentdiscounttype.md

```diff
@@ -0,0 +1,39 @@
+# Commitment Discount Type
+
+A commitment-based discount is a contractual commitment in exchange for a discounted unit price. Commitment discounts are available as spend-based or resource-based discounts, depending on the CSP offering them and the particular service.
+
+This column determines whether the commitment is based on a minimum spend or minimum usage.
```

Comment:
Does this column apply for:
- Lines corresponding to commitment purchases (when a RI/SP is purchased)
- Lines corresponding to commitment usages (when a resource is used and the RI/SP discount applies)
- Both?

Reply:
nit: Follow the typical "A <column name> is..." pattern. Might be good to say reference the CommitmentDiscountId column, since this is in reference to that.

Reply:
I am not 100% sure on this, but I would imagine it only applies where the ChargeType is Purchase - ie. the record where you bought the ComittmentDiscount.  I guess this depends on which column records the unit price discount that has been in given in exchange for the comittment.

Reply:
I would expect this to be on the usage records as well.

Reply:
Ok, but it would only apply where the CommitmentDiscountId is not null

Reply:
Could a ChargeType be a Purchase of something that is not a commitment discount? I believe ChargeType can be Purchase for Marketplace. In this case, CommitmentDiscountId would not apply.

Furthemore, a CommitmentDiscountType could also apply where the ChargeType is UnusedCommittment (in Azure, UnusedCommittment is actually either UnusedReservation or UnusedSavingsPlan).

The specification says later "When a charge is not associated with a commitment-based discount, the column MUST be null." So, maybe the column CommitmentDiscountType must not be null if and only if the rows related to  a commitment  discount, whatever its ChargeType. Or maybe we need to have a vision or direction on structuring dimensions ChargeType (and PricingModel to me), which drives the behavior of other dimensions like CommitmentDiscountType and CommitmentDiscountId.

Reply:
All `CommitmentDiscount*` columns should be specified when the charge (any type of charge) is related to a commitment-based discount.

We haven't closed unused commitment handling yet. I forwarded you an invite in case you can make it :-) 

+1 on your last paragraph. `PricingModel` convos are pending. If you have thoughts, we could setup time to chat!